src/MICore/JsonLaunchOptions.cs

```diff
@@ -0,0 +1,392 @@
+﻿// © Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+
+namespace MICore.Json.LaunchOptions
+{
+    public abstract partial class BaseOptions
+    {
+        /// <summary>
+        /// Semicolon separated list of directories to use to search for .so files. Example: "c:\dir1;c:\dir2".
+        /// </summary>
+        [JsonProperty("additionalSOLibSearchPath", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string AdditionalSOLibSearchPath { get; set; }
+
+        /// <summary>
+        /// Full path to program executable.
+        /// </summary>
+        [JsonProperty("program")]
+        public string Program { get; set; }
+
+        /// <summary>
+        /// The type of the engine. Must be "cppdbg".
+        /// </summary>
+        [JsonProperty("type", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string Type { get; set; }
```

Comment:
@gregg-miskelly I added validation for this as we're going to require cppdbg to be the type.